netlogo-gui/src/main/prim/threed/_breedat.scala

```diff
@@ -3,39 +3,38 @@
 package org.nlogo.prim.threed
 
 import org.nlogo.core.AgentKind
-import org.nlogo.agent.{ Agent3D, ArrayAgentSet, Turtle }
-import org.nlogo.api.{ AgentException}
+import org.nlogo.agent.{Agent3D, AgentSet, AgentSetBuilder, ArrayAgentSet, Turtle}
+import org.nlogo.api.AgentException
 import org.nlogo.core.Syntax
-import org.nlogo.nvm.{ Context, Reporter }
+import org.nlogo.nvm.{Context, Reporter}
 
 class _breedat(breedName: String) extends Reporter {
   def this() = this(null)
 
   override def toString =
     super.toString + ":" + breedName
-  override def report(context: Context): ArrayAgentSet = {
+  override def report(context: Context): AgentSet = {
     val dx = argEvalDoubleValue(context, 0)
     val dy = argEvalDoubleValue(context, 1)
     val dz = argEvalDoubleValue(context, 2)
     val patch =
       try context.agent.asInstanceOf[Agent3D].getPatchAtOffsets(dx, dy, dz)
       catch {
         case e: AgentException =>
-          return new ArrayAgentSet(AgentKind.Turtle, 0, false)
+          return AgentSet.fromArray(AgentKind.Turtle, Array())
       }
     if (patch == null)
-      new ArrayAgentSet(AgentKind.Turtle, 0, false)
+      AgentSet.fromArray(AgentKind.Turtle, Array())
```

Comment:
Do you think it would make sense to add an `AgentSet.empty` method to name this case explicitly? If `ArrayAgentSet` is immutable, `AgentSet.empty` could even return pre-allocated AgentSets.